### ISSUE TITLE
Microsoft.ApplicationInsights.PerfCounterCollector uses Microsoft.Extensions.Caching.Memory 1.0.0 which has CVE-2024-43483

### ISSUE THREAD

**Issue by 304NotModified** [microsoft/ApplicationInsights-dotnet#2923]
The latest version of Microsoft.ApplicationInsights.PerfCounterCollector (2.22), used by the lasted version of Microsoft.ApplicationInsights.AspNetCore (2.22) and Microsoft.ApplicationInsights.WorkerService (2.22) has a CVE. 

See CVE-2024-43483 https://github.com/advisories/GHSA-qj66-m88j-hmgj

![Image](https://github.com/user-attachments/assets/110faee5-e7e0-4a51-b9e0-5f1849861fdf)

**Comment by 304NotModified** [comment]
related https://github.com/microsoft/ApplicationInsights-dotnet/issues/2919

**Comment by pregress** [comment]
Can this please be fixed asap?

**Comment by 304NotModified** [comment]
FYI,

I was trying to upgrade Microsoft.Extensions.Caching.Memory to 8.0.1, but this gives some more work.
- System.Diagnostics.DiagnosticSource needs to be updated then. That's not bad, as the used version (5.0.0) is also deprecated on nuget.org (see https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/5.0.0)
- When bumping  System.Diagnostics.DiagnosticSource to 8.0.1, we cannot build anymore for .NET 4.5 and 4.6.0. We need a bump to 4.6.2. Is this OK? 4.5 and 4.6.0 are also not maintainted/supported anymore, see https://endoflife.date/dotnetfx. Is it OK to change the targetframeworks and the pipeline for that?
- For System.Diagnostics.DiagnosticSource has also a Redfield in the .props: https://github.com/microsoft/ApplicationInsights-dotnet/blob/c442ad30783d48b211650f7ba699735f175d6437/Directory.Build.props#L17 - I don't know wat Redfield means. Can this be removed? 

There is a some more work to do here than just updating the version. 
I only can continue on my PR after some approvals/questions answered.

Maybe it's better that MS will do this update.

**Comment by pregress** [comment]
I agree, it's not a simple fix, but Since this is a Microsoft.* package, maybe it should use the standard version strategy aka, follow the dotnet versioning, since there are dependencies on packages that follow that version strategy.

**Comment by 304NotModified** [comment]
I see that https://github.com/microsoft/ApplicationInsights-dotnet/issues/2912 in on the next milestone  (2.23), so I will close this as duplicate

**Comment by TimothyMothra** [comment]
This library does not appear to be affected by CVE-2024-43483.

The version of our dependency is outside the affected versions listed.

**Comment by 304NotModified** [comment]
Unfortunately Nexus IQ / SonarType IQ thinks this is CVE-2024-43483...

![Image](https://github.com/user-attachments/assets/e43d5c9d-2d2c-43ef-9fcd-d41c8bf8e371)

**Comment by pregress** [comment]
> Unfortunately Nexus IQ / SonarType IQ thinks this is [CVE-2024-43483](https://github.com/advisories/GHSA-qj66-m88j-hmgj)...
> 
> ![Image](https://github.com/user-attachments/assets/e43d5c9d-2d2c-43ef-9fcd-d41c8bf8e371)
> 
> ![Image](https://github.com/user-attachments/assets/4f819525-44e2-4b94-b7c2-569728c1b8f3)

Same for https://dependencytrack.org/

**Comment by TimothyMothra** [comment]
I can't speak for third party tools. The github advisory (https://github.com/advisories/GHSA-qj66-m88j-hmgj) and the dotnet announcement (https://github.com/dotnet/announcements/issues/327) are the sources of truth and both list the exact versions affected.

I personally ran our internal scanning tools yesterday and it reported no issues with the v1.0.0 package.

**Comment by 304NotModified** [comment]
So maybe the third party are wrong (not the first time). But maybe it would be nice we would not be dependent of a dependency of years old. I'm doubting it that version has even security support? But ok, that is #2951
